controller/Proctoring.php

```diff
@@ -0,0 +1,141 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA;
+ *
+ *
+ */
+
+namespace oat\taoProctoring\controller;
+
+use \common_session_SessionManager as SessionManager;
+use \core_kernel_classes_Resource;
+use oat\taoProctoring\helpers\Delivery;
+use oat\taoProctoring\helpers\TestCenter;
+use \DateTime;
+
+/**
+ *
+ * @author Open Assessment Technologies SA
+ * @package taoProctoring
+ * @license GPL-2.0
+ *
+ */
+class Proctoring extends \tao_actions_CommonModule
+{
+    protected $currentTestCenter = null;
+    protected $currentDelivery   = null;
+
+    /**
+     * Get the requested test center resource
+     * Use this to identify which test center is currently being selected buy the proctor
+     *
+     * @return core_kernel_classes_Resource
+     * @throws \common_Exception
+     */
+    protected function getCurrentTestCenter()
+    {
+        if (is_null($this->currentTestCenter)) {
+            if($this->hasRequestParameter('testCenter')){
+
+                //get test center resource from its uri
+                $testCenterUri           = $this->getRequestParameter('testCenter');
+                $this->currentTestCenter = TestCenter::getTestCenter($testCenterUri);
+            }else{
+                //@todo use a better exception
+                throw new \common_Exception('no current test center');
+            }
+            
+        }
+        return $this->currentTestCenter;
+    }
+
+    /**
+     * Get the requested delivery resource
+     * Use this to identify which delivery is currently being selected buy the proctor
+     * 
+     * @return core_kernel_classes_Resource
+     * @throws \common_Exception
+     */
+    protected function getCurrentDelivery()
+    {
+        if (is_null($this->currentDelivery)) {
+            if($this->hasRequestParameter('delivery')){
+
+                //get test center resource from its uri
+                $deliveryUri           = $this->getRequestParameter('delivery');
+                $this->currentDelivery = Delivery::getDelivery($deliveryUri);
+            }else{
+                //@todo use a better exception
+                throw new \common_Exception('no current delivery');
+            }
+        }
+        return $this->currentDelivery;
+    }
+
+    /**
+     * Main method to render a view for all proctoring related controller actions
+     * 
+     * @param string $cssClass
+     * @param array $data
+     * @param array $breadcrumbs
+     * @param String $template
+     */
+    protected function composeView($cssClass, $data = array(), $breadcrumbs = array(), $template = '')
```

Comment:
If I understand well, this function let's you aggregate all your controller's data and render the main view with unspecified _data_. But it also helps you to retrieve only the breadcrumb data when you're in a jQuery ajax context but using the same endpoint ? If yes, please separate those 2 uses contexts into different functions and even endpoints (you can still have the same method that populate the 2 endpoints)